go-controller/hack/update-codegen.sh

```diff
@@ -57,3 +56,6 @@ done
 echo "Generating CRDs"
 mkdir -p _output/crds
 controller-gen crd paths=./pkg/crd/... output:crd:dir=_output/crds
+echo "Editing egressFirewall CRD"
+sed -i -e ':begin;$!N;s/.*metadata:\n.*type: object/        metadata: \n          type: object\n          properties:\n            name:\n              type: string\n              pattern: ^default$/;tbegin;P;D' \
```

Comment:
I don't understand this. Why is this added here and not in `pkg/crd/egressfirewall/v1/types.go`? Is it because there's no supported `kubebuilder` tag that ensures this? 

In any case, I think you will need to add the output of your CRD to the git tree so that people can use it as is without re-generating it. If you do that you can fix this in the CRD directly and have any new changes to the CRD tracked, and reviewed.

Reply:
We discussed this on slack. I want to restrict the name of the egressFirewall object to 'default' so that only one can be present in the namespace this is the way we figured how to do that without writting a webhook. Since I cannot directly place validation on the name since it comes from the metav1.ObjectTypes scripting it into the CRD is the best way to prevent manual editing of the CRD after its generation.